tools/AutoTuner/src/autotuner/distributed.py

```diff
@@ -360,42 +361,22 @@ def read_tune_pbt(name, this):
     return config, sdc_file, fr_file
 
 
-def parse_flow_variables():
+def parse_tunable_variables():
     """
-    Parse the flow variables from source
-    - Code: Makefile `vars` target output
-
+    Parse the tunable variables from variables.yaml
     TODO: Tests.
-
-    Output:
-    - flow_variables: set of flow variables
     """
     cur_path = os.path.dirname(os.path.realpath(__file__))
-
-    # first, generate vars.tcl
-    makefile_path = os.path.join(cur_path, "../../../../flow/")
-    initial_path = os.path.abspath(os.getcwd())
-    os.chdir(makefile_path)
-    result = subprocess.run(["make", "vars", f"PLATFORM={args.platform}"])
-    if result.returncode != 0:
-        print(f"[ERROR TUN-0018] Makefile failed with error code {result.returncode}.")
-        sys.exit(1)
-    if not os.path.exists("vars.tcl"):
-        print(f"[ERROR TUN-0019] Makefile did not generate vars.tcl.")
-        sys.exit(1)
-    os.chdir(initial_path)
-
-    # for code parsing, you need to parse from both scripts and vars.tcl file.
-    pattern = r"(?:::)?env\((.*?)\)"
-    files = glob.glob(os.path.join(cur_path, "../../../../flow/scripts/*.tcl"))
-    files.append(os.path.join(cur_path, "../../../../flow/vars.tcl"))
-    variables = set()
-    for file in files:
-        with open(file) as fp:
-            matches = re.findall(pattern, fp.read())
-        for match in matches:
-            for variable in match.split("\n"):
-                variables.add(variable.strip().upper())
+    vars_path = os.path.join(cur_path, "../../../../flow/scripts/variables.yaml")
+
+    # Read from variables.yaml and get variables with tunable = 1
+    with open(vars_path) as file:
+        try:
+            result = yaml.safe_load(file)
+        except yaml.YAMLError as exc:
+            print("[ERROR TUN-0018] Error parsing variables.yaml.")
```

Comment:
Remove "try/except". This takes actionable informaiton(details of yaml error) and converts it into something that isn't actionable.